config/config_test.go

```diff
@@ -1220,22 +1221,70 @@ func TestConfigRobotRevision(t *testing.T) {
 	test.That(t, cfg.Revision, test.ShouldEqual, "rev1")
 }
 
-func TestConfigMarshalUnMarshal(t *testing.T) {
-	c := config.Config{
-		MaintenanceConfig: &config.MaintenanceConfig{
-			SensorName:            "SensorName",
-			MaintenanceAllowedKey: "Key",
-		},
+func TestConfigJSONMarshalRoundtrip(t *testing.T) {
+	type testcase struct {
+		name     string
+		c        config.Config
+		expected config.Config
```

Comment:
i kept an `expected` field to match the original test, but so far it seems like we expect a `config.Config -> JSON -> config.Config` round trip to result in the same loaded config, so we could consider removing this at some point. I know `UnmarshalJSON` makes adjustments in some cases, but maybe we should architect away from this pattern.